x-pack/plugins/ingest_manager/server/services/index.ts

```diff
@@ -24,12 +24,38 @@ export interface ESIndexPatternService {
  * A service that provides exported functions that return information about an Agent
  */
 export interface AgentService {
+  /**
+   * Get an Agent by id
+   */
+  getAgent(soClient: SavedObjectsClientContract, agentId: string): Promise<Agent>;
+  /**
+   * Authenticate an agent with access toekn
+   */
+  authenticateAgentWithAccessToken(
+    soClient: SavedObjectsClientContract,
+    request: KibanaRequest
+  ): Promise<Agent>;
   /**
    * Return the status by the Agent's id
-   * @param soClient
-   * @param agentId
    */
   getAgentStatusById(soClient: SavedObjectsClientContract, agentId: string): Promise<AgentStatus>;
+  /**
+   * List agents
+   */
+  listAgents(
+    soClient: SavedObjectsClientContract,
+    options: {
+      page: number;
+      perPage: number;
+      kuery?: string;
```

Comment:
@nchaulet thanks for making this change. So we must page through the data? What is the relationship between showInactive and kuery? And do you have  an example of the kuery to get a the unenrolled agent that have or had an Endpoint Integration?

Reply:
the `showInactive` allow to show inactive agent (unenrolled), kuery allow to filter on other fields

For the kuery to get unenrolled agent that have an endpoint integration it will be something like this (if this get merged https://github.com/elastic/kibana/pull/69731) 
```
listAgents(soClient, {
  showInactive: true,
  kuery: `fleet-agents.package:endpoint`
});
```

if this is not merged you will have to provide the config id 
```
listAgents(soClient, {
  showInactive: true,
  kuery: `fleet-agents.config_id:config_with_endpoint_1 or fleet-agents.config_id:config_with_endpoint_1`
});
```

Also I am working on adding an `unrolled_at` field so you can filter by date after.

Reply:
I prefer the first one. I can start coding against the API once you merge.